tools/test.pl

```diff
@@ -10597,6 +10598,43 @@ sub gen_hash
 
     $tmp_hash = sprintf ('$fvde$%d$%d$%s$%d$%s', $Z_PK, length ($salt_bin), unpack ("H*", $salt_bin), $iterations, unpack ("H*", $blob_bin));
   }
+  elsif ($mode == 18400)
+  {
+    # Complex hash string with IV, salt, checksum, ciphertext and length
+    # Do not use salt_buf, do everything here in python
+    my $python_code = <<"END_CODE";
```

Comment:
do we really need python code here?
I didn't find anything that can't be done with perl directly in the code block below. Maybe I'm missing something.
Please explain why you use python here

Reply:
You are right, we don't need python code here. That's the just code I used while writing the kernel, and since I am not a perl monk I wrote it in python. But I have to rewrite this anyway, because right now there is no way to properly use the verify function.

Btw the last time I commited python code to the test suite was because of Streebog hashing. Didn't find anything on CPAN for that either.

Reply:
yeah, I figured that out. so my idea would be to convert the code of -m 11900 = PBKDF2-HMAC-MD5 to python too such that we only have the python dependency. Do you think that makes sense? hashlib.pbkdf2_hmac in python2 probably ?
Furthermore, the base64 conversion of the words/salts (https://github.com/hashcat/hashcat/blob/master/tools/test.pl#L8226-L8227) should be done also for the python code. We do this to avoid any code injection and also to avoid any strange/special characters as input resulting in strange behaviour (this should be pretty easy to do in python too, just encode in perl and decode the strings in python).
we should also update tools/install_modules.sh if we need any additional/special python modules (btw: should we rename the file "install_modules.sh" into something that explains that this is needed for tests ? (install_test_modules.sh ? test_install_modules.sh ?) 

BTW: I do not think everything I mentioned here should be done within this commit/PR, the PHP->python could/should be a separate PR/issue (it's just something I noticed here and wanted to mention and ask if python is really the way to go here)

Reply:
Done! :D